include/picongpu/plugins/transitionRadiation/TransitionRadiation.hpp

```diff
@@ -449,6 +528,204 @@ namespace picongpu
+                        ++index_direction)
+                    {
+                        // theta
+                        const int i = index_direction / parameters::nPhi;
+                        // phi
+                        const int j = index_direction % parameters::nPhi;
+
+                        for(unsigned int k = 0; k < transitionRadiation::frequencies::nOmega; ++k)
+                        {
+                            dataBox({static_cast<int>(i), static_cast<int>(j), static_cast<int>(k)})
+                                = static_cast<float_X>(
+                                    theTransRad[index_direction * transitionRadiation::frequencies::nOmega + k]);
+                        }
+                    }
+
+                    std::stringstream filename;
+                    filename << filenamePrefix << "_%T." << fileExtension;
+
+                    ::openPMD::Series series(filename.str(), ::openPMD::Access::CREATE);
+
+                    ::openPMD::Extent extent
+                        = {static_cast<unsigned long int>(transitionRadiation::frequencies::nOmega),
+                           static_cast<unsigned long int>(parameters::nPhi),
+                           static_cast<unsigned long int>(parameters::nTheta)};
+                    ::openPMD::Offset offset = {0, 0, 0};
+                    ::openPMD::Datatype datatype = ::openPMD::determineDatatype<float_X>();
+                    ::openPMD::Dataset dataset{datatype, extent};
+
+                    auto mesh = series.iterations[currentStep].meshes["transitionradiation"];
+
+                    mesh.setAxisLabels(std::vector<std::string>{"omega index", "phi index", "theta index"});
+                    mesh.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    mesh.setGridUnitSI(1);
+                    mesh.setGridSpacing(std::vector<double>{1, 1, 1});
+                    mesh.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+                    mesh.setAttribute<float_X>("foilPosition", foilPositionSI);
+
+                    mesh.setUnitDimension(std::map<::openPMD::UnitDimension, double>{
+                        {::openPMD::UnitDimension::L, 2.0},
+                        {::openPMD::UnitDimension::M, 1.0},
+                        {::openPMD::UnitDimension::T, -1.0}});
+
+                    auto transitionRadiation = mesh[::openPMD::RecordComponent::SCALAR];
+                    transitionRadiation.resetDataset(dataset);
+
+                    transitionRadiation.setUnitSI(
+                        SI::ELECTRON_CHARGE_SI * SI::ELECTRON_CHARGE_SI
+                        * (1.0 / (4 * PI * SI::EPS0_SI * PI * PI * SI::SPEED_OF_LIGHT_SI)));
+
+                    auto sharedDataPtr = std::shared_ptr<float_X>{buffer->getPointer(), [](auto const*) {}};
+
+                    transitionRadiation.storeChunk(sharedDataPtr, offset, extent);
```

Comment:
See comment above, I recommend changing this either to (1) ideally a Span-based storeChunk(), (2) as a slightly weaker optimization a unique-ptr based storeChunk(), (3) or if those don't work a raw-pointer based storeChunk().